app/src/main/java/org/wikipedia/login/LoginActivity.java

```diff
@@ -233,11 +231,7 @@ private void doLogin() {
         return new LoginClient.LoginCallback() {
             @Override
             public void success(@NonNull LoginResult result) {
-                if (!progressDialog.isShowing()) {
-                    // no longer attached to activity!
-                    return;
-                }
-                progressDialog.dismiss();
+                showProgressBar(false);
```

Comment:
Note: this no longer checks if the activity is still alive, which might lead to unexpected behavior, including crashing. See my follow-up.